src/UserSpaceInstrumentation/Trampoline.h

```diff
@@ -115,17 +115,19 @@ struct RelocatedInstruction {
 [[nodiscard]] uint64_t GetMaxTrampolineSize();
 
 // Creates a trampoline for the function at `function_address`. The trampoline is built at
-// `trampoline_address`. The trampoline will call `payload_address` with `function_address` as a
-// parameter. `function` contains the beginning of the function (kMaxFunctionPrologueBackupSize
-// bytes or less if the function shorter). `capstone_handle` is a handle to the capstone
-// disassembler library returned by cs_open. The function returns an error if it was not possible to
-// instrument the function. For details on that see the comments at AppendRelocatedPrologueCode. If
-// the function is successful it will insert an address pair into `relocation_map` for each
-// instruction it relocated from the beginning of the function into the trampoline (needed for
-// moving instruction pointers away from the overwritten bytes at the beginning of the function,
-// compare MoveInstructionPointersOutOfOverwrittenCode below). The return value is the address of
-// the first instruction not relocated into the trampoline (i.e. the address the trampoline jump
-// back to).
+// `trampoline_address`. The trampoline will call `entry_payload_function_address` with
+// `return_address` and a function id as a parameter. The function id is written into the
+// trampoline by `InstrumentFunction`. This is necessary since the function id is not stable accross
+// multiple profiling runs.
+// `function` contains the beginning of the function (kMaxFunctionPrologueBackupSize bytes or less
```

Comment:
Unrelated to this change, but `function` could be renamed to something more expressive such as `function_prologue_backup`.

Reply:
yep, agreed, naming is not great.